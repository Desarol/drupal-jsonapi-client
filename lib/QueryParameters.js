"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * Array.prototype.flat() is not supported in IE.
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat
 */

/* eslint-disable */
if (!Array.prototype.flat) {
  Array.prototype.flat = function (depth) {
    var flattend = [];

    (function flat(array, depth) {
      var _iteratorNormalCompletion = true;
      var _didIteratorError = false;
      var _iteratorError = undefined;

      try {
        for (var _iterator = array[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
          var el = _step.value;

          if (Array.isArray(el) && depth > 0) {
            flat(el, depth - 1);
          } else {
            flattend.push(el);
          }
        }
      } catch (err) {
        _didIteratorError = true;
        _iteratorError = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion && _iterator.return != null) {
            _iterator.return();
          }
        } finally {
          if (_didIteratorError) {
            throw _iteratorError;
          }
        }
      }
    })(this, Math.floor(depth) || 1);

    return flattend;
  };
}
/* eslint-enable */


var QueryParameters =
/*#__PURE__*/
function () {
  function QueryParameters(queryParameters) {
    _classCallCheck(this, QueryParameters);

    this.queryParameters = queryParameters;
  }

  _createClass(QueryParameters, [{
    key: "toString",
    value: function toString(depth) {
      return this.queryParameters.flat(depth).map(function (item) {
        return !!item && item.query ? item.query() : item;
      }).flat(depth).filter(function (item) {
        return !!item;
      }).join('&');
    }
  }]);

  return QueryParameters;
}();

exports.default = QueryParameters;