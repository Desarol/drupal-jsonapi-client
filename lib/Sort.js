"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SortDirection = void 0;

/**
 * Based on: https://www.drupal.org/docs/core-modules-and-themes/core-modules/jsonapi-module/sorting
 */
const SortDirection = {
  DESCENDING: 'DESC',
  ASCENDING: 'ASC'
};
exports.SortDirection = SortDirection;

class Sort {
  constructor({
    identifier,
    path,
    direction = SortDirection.ASCENDING
  }) {
    this.identifier = identifier;
    this.path = path;
    this.direction = direction;
  }
  /* eslint-disable prefer-template */


  query() {
    return [`sort[${this.identifier}][path]=${this.path}`, this.direction === SortDirection.DESCENDING ? `sort[${this.identifier}][direction]=${this.direction}` : ''];
  }
  /* eslint-enable prefer-template */


}

exports.default = Sort;