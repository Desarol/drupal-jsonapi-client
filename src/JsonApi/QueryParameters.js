/**
 * Array.prototype.flat() is not supported in IE.
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat
 */

/* eslint-disable */
if (!Array.prototype.flat) {
  Array.prototype.flat = function(depth) {
    var flattend = [];
    (function flat(array, depth) {
      for (let el of array) {
        if (Array.isArray(el) && depth > 0) {
          flat(el, depth - 1); 
        } else {
          flattend.push(el);
        }
      }
    })(this, Math.floor(depth) || 1);
    return flattend;
  };
}
/* eslint-enable */

export default class QueryParameters {
  constructor(queryParameters) {
    this.queryParameters = queryParameters
  }

  toString(depth) {
    return this.queryParameters
      .flat(depth)
      .map(item => (!!item && item.query ? item.query() : item))
      .flat(depth)
      .filter(item => !!item)
      .join('&')
  }
}
