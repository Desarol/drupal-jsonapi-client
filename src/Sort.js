/**
 * Based on: https://www.drupal.org/docs/core-modules-and-themes/core-modules/jsonapi-module/sorting
 */

export const SortDirection = {
  DESCENDING: 'DESC',
  ASCENDING: 'ASC',
}

export default class Sort {
  constructor({
    identifier,
    path,
    direction = SortDirection.ASCENDING,
  }) {
    this.identifier = identifier
    this.path = path
    this.direction = direction
  }

  /* eslint-disable prefer-template */
  query() {
    return ([
      `sort[${this.identifier}][path]=${this.path}`,
      this.direction === SortDirection.DESCENDING ? `sort[${this.identifier}][direction]=${this.direction}` : '',
    ])
  }
  /* eslint-enable prefer-template */
}
