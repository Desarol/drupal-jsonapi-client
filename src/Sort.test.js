import QueryParameters from './QueryParameters'
import Sort from './Sort'

describe('Filter', () => {
  it('serializes sort to query', () => {
    expect((new QueryParameters([
      new Sort({
        identifier: 'sort-created',
        path: 'created',
      }),
    ])).toString(10)).toMatchSnapshot()
  })
})
