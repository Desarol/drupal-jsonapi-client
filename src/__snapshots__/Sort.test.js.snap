// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter serializes sort to query 1`] = `"sort[sort-created][path]=created"`;
